sql/sql_select.cc

```diff
@@ -21806,12 +21806,22 @@ bool create_internal_tmp_table(TABLE *table, KEY *keyinfo,
   if (share->keys)
   {						// Get keys for ni_create
     bool using_unique_constraint=0;
+    uint len= keyinfo->key_length>1?keyinfo->key_length:1;
+
```

Comment:
This doesn't seem to make any sense.  key_length is some length in bytes, isn't it?

I am trying this example:

```
create table t11 (
  a int, 
  b blob
);
insert into t11
select seq, seq from seq_1_to_100;

select distinct a,b from t11;
```

I get a breakpoint on this line and I see `keyinfo->key_length=4` . 

The code below will allocate 4 MARIA_KEYDEF structures:

```
    keydef= (MARIA_KEYDEF*) alloc_root(&table->mem_root, sizeof(*keydef) * len);
```

It is not clear why.